File: 0263.丑数/0263-丑数.java
class Solution {
    public boolean isUgly(int num) {
        if (num <= 0)
            return false;
        if (num == 1 || num == 2 || num == 3 || num == 5)
            return true;
        if (num % 2 == 0)
            return isUgly(num / 2);
        if (num % 3 == 0)
            return isUgly(num / 3);
        if (num % 5 == 0)
            return isUgly(num / 5);
        return false;
    }
}

File: 0263.丑数/0263-丑数.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 0:
            return False 
        while not num % 2:
            num /= 2
        while not num % 3:
            num /= 3
        while not num % 5:
            num /= 5
        return num == 1